Upgrade airbase version to 46.

kokosing opened the pull request:
Upgrade airbase version to 46.

 - As new airbase comes with new version of testng some tests had to be
refactored as they started to fail.
 - Upgrade also airlift, discover-server and resolver to versions which
 also are using airbase at version 46, as with older version there were
 a dependency conflict.

electrum commented on presto-main/src/test/java/com/facebook/presto/util/Numbers.java:
```diff
@@ -0,0 +1,31 @@
+        return number.equals(Double.NaN) || number.equals(Float.NaN);
+    }
+
+    private Numbers()
```
Put the constructor at the top of the class and format like this:
```
private Numbers() {}

electrum commented on presto-main/src/test/java/com/facebook/presto/util/Numbers.java:
```diff
@@ -0,0 +1,31 @@
+ */
+package com.facebook.presto.util;
+
+public class Numbers
```
Utility classes should be final

electrum commented:
Commit titles should not end with a period (they are not always sentences)

kokosing commented:
Rebased, comments were addressed. I removed SNAPSHOT version of airbase. To make this patch be eligible for merge:
 - https://github.com/airlift/airlift/pull/360 needs to be merged and then airlift  released
 - https://github.com/airlift/resolver/pull/10 needs to be merged and then resolver released
 - https://github.com/airlift/discovery/pull/38 needs to be merged and then discovery  released

kokosing commented:
Rebased, SNAPSHOT versions of dependencies removed.

martint commented:
Thanks, I'll merge it after we're done cutting the current release.

martint commented:
Merged, thanks!
